network/router_test.go

```diff
@@ -61,6 +61,13 @@ func testRouter(t *testing.T, fac routerFactory) {
 	case <-time.After(500 * time.Millisecond):
 		t.Fatal("TcpHost should have returned from Run() by now")
 	}
+
+	require.Nil(t, h.errorHappened)
+	errHandler := func(remote *ServerIdentity) {
+		//nothing to do here
+	}
+	h.SetErrorHandler(errHandler)
+	require.NotNil(t, h.errorHappened)
```

Comment:
Test also whether the thing really works or not. If an a connection drops out, do you receive something on the handler?